libs/@blockprotocol/graph/src/codegen/postprocess/generate-block-link-target-aliases.ts

```diff
@@ -0,0 +1,50 @@
+import { mustBeDefined } from "../../shared/util/must-be-defined.js";
+import {
+  typedEntries,
+  typedValues,
+} from "../../shared/util/typed-object-iter.js";
+import { PostprocessContext } from "../context/postprocess.js";
+import { entityDefinitionNameForEntityType } from "../shared.js";
+
+export const generateBlockLinkTargetAliases = (
+  context: PostprocessContext,
+): void => {
+  context.logDebug("Generating block entity types to generated files");
+
+  for (const [fileName, { blockEntity }] of typedEntries(
+    context.parameters.targets,
+  )) {
+    if (blockEntity) {
+      const type = mustBeDefined(context.allTypes[blockEntity]);
+      const typeName = entityDefinitionNameForEntityType(type.title);
+
+      /* @todo - make this more robust */
+      const entityLinkAndTargetsIdentifier = `${typeName}OutgoingLinksAndTargets`;
+
+      if (
+        typedValues(context.filesToDefinedIdentifiers).find(
+          (definedIdentifier) =>
+            definedIdentifier.has(entityLinkAndTargetsIdentifier),
+        )
+      ) {
+        context.logDebug(
+          `Generating block entity links and targets alias ${typeName} for file ${fileName}`,
+        );
+
+        const identifier = "BlockEntityOutgoingLinksAndTargets";
+
+        const blockEntityLinkAndTargetTypesFragment = `\nexport type ${identifier} = ${entityLinkAndTargetsIdentifier}\n`;
```

Comment:
We won't run into clasing identifiers because we only allow one block entity type per file, right?

Reply:
Correct, and that's checked as part of validating the `CodegenParameters`, and we only hoist "locally importable" identifiers to `shared.ts` so these will be unique per generated file if they're defined.